File: app/src/main/java/com/malakezzat/re7letelkalemat/View/Interfaces/AuthView.java
package com.malakezzat.re7letelkalemat.View.Interfaces;

public interface AuthView {
    void showLoading();
    void hideLoading();
    void showToast(String message);
    void navigateToHome(String email);
    void setEmailError(String error);
    void setPasswordError(String error);
}

File: app/src/main/java/com/malakezzat/re7letelkalemat/View/Interfaces/AuthView.kt
package com.malakezzat.re7letelkalemat.View.Interfaces

interface AuthView {
    open fun showLoading()
    open fun hideLoading()
    open fun showToast(message: String?)
    open fun navigateToHome(email: String?)
    open fun setEmailError(error: String?)
    open fun setPasswordError(error: String?)
}